sql/core/src/main/scala/org/apache/spark/sql/execution/joins/CartesianProduct.scala

```diff
@@ -54,7 +62,10 @@ case class CartesianProduct(left: SparkPlan, right: SparkPlan) extends BinaryNod
       val joinedRow = new JoinedRow
       iter.map { r =>
         numOutputRows += 1
-        joinedRow(r._1, r._2)
+        buildSide match {
```

Comment:
Nit: Move the `buildSide` pattern match out of the `map { }`, probably can reduce some overhead of runtime matching.